wp-content/themes/workingnyc/src/js/modules/programs-archive.js

```diff
@@ -187,7 +227,179 @@ export default {
+       * The query argument gets the current, previous, next page
+       * So, we set Posts array with current page posts
+       */
+      
+      if(this.query.page == query.page){
+        this.$set(this.posts[query.page], 'posts', posts);
+        this.$set(this.posts[query.page], 'headers', Object.freeze(headers));
+      }
+
+      // If there are no posts, pass along to the error handler.
+      if (!Array.isArray(data))
+        this.error({error: data, query: query});
+
+      this.$set(this, 'init', true);
+
+      if(this.headers.pages<=6){
+        this.firstPage = false;
+        this.lastPage = false;
+        this.totalPages = this.headers.pages;
+      }
+      else{
+        this.firstPage = true;
+        this.lastPage = true;
+        this.totalPages = [2,3,4,5]
+      } 
+    },
+
+    /**
+     * Fetch the posts of updated page
+     */
+    updatePagination: function(){
+      this.scrollToTop();
+      let url = [
+        this.domain,
+        this.lang.path,
+        this.endpoints[this.type],
+        this.buildUrlQuery(this.query)
+      ].join('');
+
+      fetch(url)
+      .then(this.response)
+      .then(data => {
+        let headers = Object.assign({}, this.headers);
+        this.posts.length=1;
+        let currentPost = {
+          'posts': data,
+          'headers': headers,
+          'query': this.query,
+          'show': true
+        };
+        this.posts.push(currentPost);
+        this.setPagination();      
+      }).catch(this.error);
+    },
+
+    previousPage:function(event){
+      let currPage = this.posts[1].query.page;
+      this.$set(this.query, 'page', currPage-1);
+      this.updatePagination();
+    },
+
+    nextPagination:function(event){
+      let currPage = this.posts[1].query.page;
+      this.$set(this.query, 'page', currPage+1);
+      this.updatePagination();
+    },
+
+    immediatePage: function(event){
+      let change = parseInt(event.target.dataset.amount);
+      this.$set(this.query, 'page', change);
+      this.updatePagination();
+    },
+
+    updateQuery: function(taxonomy, terms) {
+      return new Promise((resolve) => { // eslint-disable-line no-undef
+        this.$set(this.query, taxonomy, terms);
+        this.$set(this.query, 'page', 1);
+        // hide all of the posts
+        if(this.posts.length>1){
+          this.posts.length=1;
+        }
+        resolve();
+      })
+      .then(this.wp)
+      .catch(message => {
+      });
+    },
+
+    /**
+     * Update the pagination array
+     * The maximum pages: 6
+     */
+    setPagination: function(){
+
+      /**
+       * If number of pages is less than 7
+       *  The totalPages = [1,2,3,4,5,6]
+       *  The Pagination = 1,2,3,4,5,6
+       * else
+       *  The totalPages = [2,3,4,5]
+       *  The Pagination = firstPage,2,3,4,5..lastPage
+       */
+
+      if(this.headers.pages<=6){
+        this.firstPage = false;
+        this.lastPage = false;
+        this.totalPages = this.headers.pages;
```

Comment:
I thought this.headers.pages is a number and this.totalPages is an array? Why does this statement work?

Reply:
Yes, totalPages is an array. I will update it